tapiriik/services/Aerobia/aerobia.py

```diff
@@ -0,0 +1,340 @@
+            raise APIException("Api call not succeed", user_exception=UserException(UserExceptionType.DownloadError))
+        return resp
+
+    def _refresh_token(self, record):
+        logger.info("refreshing auth token")
+        user_id, user_token = self._get_auth_data(record=record)
+        record.Authorization.update({"OAuthToken": user_token})
+
+    def _with_auth(self, record, params={}):
+        params.update({"authentication_token": record.Authorization["OAuthToken"]})
+        return params
+
+    def Authorize(self, username, password):
+        from tapiriik.auth.credential_storage import CredentialStore
+        user_id, user_token = self._get_auth_data(username=username, password=password)
+
+        secret = {
+            "Email": CredentialStore.Encrypt(username), 
+            "Password": CredentialStore.Encrypt(password)
+            }
+        authorizationData = {"OAuthToken": user_token}
+        
+        return (user_id, authorizationData, secret)
+
+    def DownloadActivityList(self, serviceRecord, exhaustive=False):
+        self._patch_user_agent()
+
+        activities = []
+        exclusions = []
+
+        fetch_dairy = lambda page=1: self._get_dairy_xml(serviceRecord, page)
+        # use first query responce to detect pagination options as well
+        dairy_xml = self._call(serviceRecord, fetch_dairy)
+
+        pagination = dairy_xml.find("pagination")
+        # New accounts have no data pages initially
+        total_pages_str = pagination.get("total_pages") if pagination else None
+        total_pages = int(total_pages_str) if total_pages_str else 1
+        
+        for page in range(2, total_pages + 2):
+            for workout_info in dairy_xml.findall("workouts/r"):
+                activity = self._create_activity(workout_info)
+                activities.append(activity)
+            
+            if not exhaustive or page > total_pages:
+                break
+            dairy_xml = self._call(serviceRecord, fetch_dairy, page)
+
+        return activities, exclusions
+
+    def _get_dairy_xml(self, serviceRecord, page=1):
+        dairy_data = requests.get(self._workoutsUrl, params=self._with_auth(serviceRecord, {"page": page}))
+        dairy_xml = etree.fromstring(dairy_data.text.encode('utf-8'))
+
+        info = dairy_xml.find("info")
+        if info.get("status") != "ok":
+            raise APIException(info.get("description"), user_exception=UserException(UserExceptionType.DownloadError))
+
+        return dairy_xml
+
+    def _create_activity(self, data):
+        activity = UploadedActivity()
+        activity.Name = data.get("name")
+        activity.StartTime = pytz.utc.localize(datetime.strptime(data.get("start_at"), "%Y-%m-%dT%H:%M:%SZ"))
+        activity.EndTime = activity.StartTime + timedelta(0, float(data.get("duration")))
+        sport_id = data.get("sport_id")
+        activity.Type = self._reverseActivityMappings[int(sport_id)] if sport_id else ActivityType.Other
```

Comment:
Might want to do `_reverseActivityMappings.get(int(sport_id), ActivityType.Other)` to account for future additions.